pkgs/tools/security/nmap/default.nix

```diff
@@ -33,9 +33,21 @@ in stdenv.mkDerivation rec {
         --replace /usr/bin/libtool ar \
         --replace 'AR="libtool"' 'AR="ar"' \
         --replace 'ARFLAGS="-o"' 'ARFLAGS="-r"'
+  '' + optionalString (stdenv.buildPlatform != stdenv.hostPlatform) ''
+    sed -i -r 's@^RANLIB\s*=\s*ranlib@RANLIB=${stdenv.cc.bintools.targetPrefix}ranlib@' liblinear/Makefile
+    sed -i -r 's@^RANLIB\s*=\s*ranlib@RANLIB=${stdenv.cc.bintools.targetPrefix}ranlib@' liblinear/blas/Makefile
+    sed -i -r 's@^RANLIB\s*=\s*ranlib@RANLIB=${stdenv.cc.bintools.targetPrefix}ranlib@' liblua/Makefile
+    sed -i -r 's@^AR\s*=\s*ar@AR=${stdenv.cc.bintools.targetPrefix}ar@'                 liblinear/Makefile
+    sed -i -r 's@^AR\s*=\s*ar@AR=${stdenv.cc.bintools.targetPrefix}ar@'                 liblinear/blas/Makefile
+    sed -i -r 's@^AR\s*=\s*ar@AR=${stdenv.cc.bintools.targetPrefix}ar@'                 liblua/Makefile
+    sed -i -r 's@^AR\s*=\s*ar@AR=${stdenv.cc.bintools.targetPrefix}ar@'                 libnetutil/Makefile.in
+    sed -i -r 's@^AR\s*=\s*ar@AR=${stdenv.cc.bintools.targetPrefix}ar@'                 libpcre/Makefile.in
+    sed -i -r 's@^AR\s*=\s*ar@AR=${stdenv.cc.bintools.targetPrefix}ar@'                 nbase/Makefile.in
+    sed -i -r 's@^AR\s*=\s*ar@AR=${stdenv.cc.bintools.targetPrefix}ar@'                 nsock/src/Makefile.in
+    sed -i -r 's@^CC\s*=\s*gcc@CC=${stdenv.cc.targetPrefix}gcc@'                        liblua/Makefile
```

Comment:
I think it would be nice to have this replacement done a bit less hardcoded, by running sed on all `Makefile`/`Makefile.in` files, with `find` or so.

Or maybe just run these sed commands over multiple files at once (`sed -i -r '...' liblinear/Makefile liblinear/blas/Makefile ...`)

Reply:
Can this be overridden with `makeFlags = [ "AR=${stdenv.cc.bintools.targetPrefix}ar" ];'` etc?

Reply:
Yes, this works, but the build fails later for the different reason (```armv7l-unknown-linux-gnueabihf-ar: invalid option -- '.'```) :(

Reply:
Weird where does these arguments come from?

Reply:
I pushed your version so we'll see in ofborg logs

Reply:
It should be ok to go with cleaned up sed solution unless you find something better.